cmd/k8s-operator/sts.go

```diff
@@ -776,45 +782,41 @@ func applyProxyClassToStatefulSet(pc *tsapi.ProxyClass, ss *appsv1.StatefulSet,
 func enableEndpoints(ss *appsv1.StatefulSet, metrics, debug bool) {
 	for i, c := range ss.Spec.Template.Spec.Containers {
 		if c.Name == "tailscale" {
-			if debug {
+			if metrics {
 				ss.Spec.Template.Spec.Containers[i].Env = append(ss.Spec.Template.Spec.Containers[i].Env,
-					// Serve tailscaled's debug metrics on on
-					// <pod-ip>:9001/debug/metrics. If we didn't specify Pod IP
-					// here, the proxy would, in some cases, also listen to its
-					// Tailscale IP- we don't want folks to start relying on this
-					// side-effect as a feature.
-					corev1.EnvVar{
-						Name:  "TS_TAILSCALED_EXTRA_ARGS",
-						Value: "--debug=$(POD_IP):9002",
-					},
+					// Serve client metrics on <pod-ip>:9001/metrics and <pod-ip>:9001/debug endpoints.
 					corev1.EnvVar{
-						Name:  "TS_DEBUG_ENDPOINT_ADDR_PORT",
-						Value: "$(POD_IP):9002",
+						Name:  "TS_METRICS_ADDR_PORT",
+						Value: "$(POD_IP):9001",
 					},
 				)
-
 				ss.Spec.Template.Spec.Containers[i].Ports = append(ss.Spec.Template.Spec.Containers[i].Ports,
 					corev1.ContainerPort{
-						Name:          "debug",
+						Name:          "metrics",
 						Protocol:      "TCP",
-						ContainerPort: 9002,
+						ContainerPort: 9001,
 					},
 				)
 			}
 
-			if metrics {
+			if debug {
 				ss.Spec.Template.Spec.Containers[i].Env = append(ss.Spec.Template.Spec.Containers[i].Env,
-					// Serve client metrics on <pod-ip>:9001/metrics and <pod-ip>:9001/debug endpoints.
+					// Serve tailscaled's debug metrics on on
+					// <pod-ip>:9002/debug/metrics. If we didn't specify Pod IP
+					// here, the proxy would, in some cases, also listen to its
+					// Tailscale IP- we don't want folks to start relying on this
+					// side-effect as a feature.
 					corev1.EnvVar{
-						Name:  "TS_METRICS_ADDR_PORT",
-						Value: "$(POD_IP):9001",
+						Name:  "TS_DEBUG_ADDR_PORT",
+						Value: "$(POD_IP):9002",
```

Comment:
We could have kept the debug port as 9001 and the new metrics port as 9002. Currently the debug port is 9001, so if someone deploys older proxy with this new operator version, they won't get the debug metrics at the same port.

Reply:
Very good point, fixed in 6c35d0d